src/lib/setGlobalConfig.ts

```diff
@@ -7,34 +10,23 @@ export default function setGlobalConfig(
   this: ModuleThis,
   _config: YahooFinanceOptions
 ): void {
-  // Instances (e.g. cookieJar) don't validate well :)
-  const { cookieJar, logger, ...config } = _config;
-
-  validateAndCoerceTypes({
-    object: config,
-    source: "setGlobalConfig",
+  const parsed = validateAndCoerceTypes({
+    data: _config,
     type: "options",
     options: this._opts.validation,
-    schemaKey: "#/definitions/YahooFinanceOptions",
+    schema: YahooFinanceOptionsSchema,
   });
+
+  // Instances (e.g. cookieJar) don't validate well :)
+  const { cookieJar, ...config } = parsed;
+
   mergeObjects(this._opts, config);
 
   if (cookieJar) {
     if (!(cookieJar instanceof ExtendedCookieJar))
       throw new Error("cookieJar must be an instance of ExtendedCookieJar");
     this._opts.cookieJar = cookieJar;
   }
-  if (logger) {
```

Comment:
I took this out (and added a test to that effect) because this is actually covered by the Typebox validation now.

Happy to revert if it's too unrelated

Reply:
no that's much better, thanks! :clap: